src/Control/Monad/Trans/Abort.hs

```diff
@@ -110,9 +110,10 @@ instance (Functor μ, Monad μ) ⇒ Bind (AbortT e μ) where
 instance Monad μ ⇒ Monad (AbortT e μ) where
   return  = AbortT . return . Right
   m >>= f = AbortT $ runAbortT m >>= either (return . Left) (runAbortT . f)
+#if !MIN_VERSION_base(4,9,0)
```

Comment:
Should be
```haskell
#if !MIN_VERSION_base(4,13,0)
  fail = AbortT . fail
#endif
#if MIN_VERSION_base(4,9,0)
instance Fail.MonadFail ...
#endif
```
otherwise `fail` will default to `error` with base >= 4.9 && <= 4.12

Reply:
Thank you, updated.